mpich: Improve package

michaelkuhn opened the pull request:
- Add missing pkg-config, libpciaccess and libxml2 dependencies
- Add missing slurm dependency for mpich@3.3:
- Add missing configure flag for libfabric

alalazo commented on var/spack/repos/builtin/packages/mpich/package.py:
```diff
@@ -158,4 +164,9 @@ def configure_args(self):
+        # Specify libfabric's path explicitly, otherwise configure might fall
+        # back to an embedded version of libfabric.
+        if 'netmod=ofi' in spec:
+            config_args.append('--with-libfabric={0}'.format(spec['libfabric'].prefix))
```
The dreaded 80 chars line hit again.

michaelkuhn replied:
This is what happens if you forget to run flake8 one time. :laughing: Thanks!

jrood-nrel commented:
Should the slurm dependency involve a variant? I've been meaning to update the mpich package since I use it with slurm with a custom package file. I've also used 3.3 without slurm though.

Also I've been meaning to make a pull request for this as well, but maybe you could add it here?
```
+    def setup_environment(self, spack_env, run_env):
+        spec = self.spec
+        # mpich2 configure fails when F90 and F90FLAGS are set
+        spack_env.unset('F90')
+        spack_env.unset('F90FLAGS')
```

michaelkuhn commented:
> Should the slurm dependency involve a variant? I've been meaning to update the mpich package since I use it with slurm with a custom package file. I've also used 3.3 without slurm though.

This is probably not what you meant but I just noticed that SLURM is only required if Hydra is enabled. I have updated the PR accordingly.

> Also I've been meaning to make a pull request for this as well, but maybe you could add it here?

Good point. I have not noticed this since recent modulefiles do not seem to set `F90`/`F90FLAGS`. I have also added this to the PR.

@alalazo The build tests fail with "The job exceeded the maximum log length, and has been terminated." Is there anything I can do from my side?

jrood-nrel commented:
> Good point. I have not noticed this since recent modulefiles do not seem to set `F90`/`F90FLAGS`. I have also added this to the PR.

Cool, thanks!

michaelkuhn commented:
I have just updated the PR to allow selecting different PMI interfaces.

alalazo commented:
> The build tests fail with "The job exceeded the maximum log length, and has been terminated."  Is there anything I can do from my side?

I see two ways to reduce the log length:

1. Reduce the verbosity of the `mpich` build, but that can trigger job termination because of 10 mins. without output
2. Install a few dependencies with apt and configure them as externals for the sake of the build. test Maybe we should try with the dependencies added by this PR if they are packaged in Ubuntu?

michaelkuhn commented:
> 1. Reduce the verbosity of the `mpich` build, but that can trigger job termination because of 10 mins. without output

Disclaimer: I have no idea about Travis. :-) Is it possible to increase the limits for single tests?

> 2. Install a few dependencies with apt and configure them as externals for the sake of the build. test Maybe we should try with the dependencies added by this PR if they are packaged in Ubuntu?

They all seem to be available in Ubuntu (at least on 18.04).
pkg-config, libpciaccess, libxml2 and libfabric should not be a problem. The slurm version in Ubuntu sadly still ships its include in a wrong directory so that configure will not find them (`/usr/include/slurm-wlm` instead of `/usr/include/slurm`).

michaelkuhn commented:
As per @jrood-nrel's suggestion, I have now added a variant to enable SLURM support (that defaults to false). SLURM is not strictly necessary even if using Hydra, so this allows disabling a heavy dependency.

From my side, this PR is now ready for review. @alalazo
